test/functional/forall/test-forall-atomic-utils.hpp

```diff
@@ -0,0 +1,80 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef __TEST_FORALL_ATOMIC_UTILS_HPP__
+#define __TEST_FORALL_ATOMIC_UTILS_HPP__
+
+#include "RAJA/RAJA.hpp"
+
+#include "test-forall-utils.hpp"
+
+using AtomicSeqExecs = camp::list< RAJA::seq_exec >;
```

Comment:
@rchen20 would it be better to share the forall execution policies that are defined in the forall utils header so they aren't repeated here?

Reply:
I'll try that, the only concern is cuda_exec, but if they already work for a block size of 256 then they ought to work for 128.

Reply:
That should work fine.

Reply:
The CUDA and HIP exec policies were changed, but there are a handful of errors with standard OpenMP so those are still using custom exec lists (documented in test-forall-atomic-utils.hpp). I will debug those in a future PR because the current combination of tests should be sufficient (@MrBurmark let me know if this is ok).

Reply:
This is a little awkward, I'd be happier if we documented that we would prefer to use the forall utils lists as well as why we can't.

Reply:
Also, a good suggestion. @rchen20 we should be able to use 'loop_exec' for the sequential forall-atomic policies.

Reply:
Marking off simd makes perfect sense, but this one:

```
              //RAJA::omp_parallel_exec<RAJA::seq_exec>
              //can work with atomics but tests not suited to omp parallel region
```

Makes me wonder why we have that for forall in the first place, it's only ever meant to be used with region really, in this setup that policy would run the loop redundantly in every thread, which for some tests should probably cause errors...

Reply:
@trws good point. What if we had an omp_for and omp_for_nowait policy types that we could pass in as the template parameter for omp_parallel_exec? Not sure that makes sense or is useful.

Reply:
Makes perfect sense, and we have those in fact: `for`, `for_nowait` and `for_static` [over here](https://github.com/LLNL/RAJA/blob/develop/include/RAJA/policy/openmp/policy.hpp#L65).

That's actually how the parallel_for_exec policy is defined:

```c++
struct omp_parallel_for_exec : omp_parallel_exec<omp_for_exec> {
};
```

Reply:
Right. I skimmed right past those. I think this usage still seems confusing with the omp_parallel_exec on the outside. IMO, it's clearer and more idiomatic to use our openmp region construct on the outside.

Reply:
I completely agree, that's really what it's for right?  That's part of why I was so surprised to see `parallel_exec<seq_exec>` on a forall though, it's technically valid but only if you want to execute the same loop redundantly in some number of threads.

Reply:
Totally agree. I think we should remove this case from the tests. We can test the outer parallel region with inner omp exec policy in the region tests we have yet to convert. @rchen20 and @mdavis36 do you agree?  If so, I can remove these in a future PR.

Reply:
If we want to have it in the tests it might be good to have a test specifically for it, in case someone wants that usage, but I'd vote for pulling it out of the general test list for otherwise workshared loops.